src/Core/Python/PythonDatatypeConverter.cc

```diff
@@ -34,30 +34,55 @@
 #pragma warning( disable: 4244 )
 #endif
 
-#include <Core/Python/PythonDatatypeConverter.h>
-#include <Core/Datatypes/SparseRowMatrix.h>
 #include <Core/Datatypes/DenseMatrix.h>
-#include <Core/Datatypes/String.h>
 #include <Core/Datatypes/Legacy/Field/Field.h>
+#include <Core/Datatypes/MatrixTypeConversions.h>
+#include <Core/Datatypes/SparseRowMatrix.h>
+#include <Core/Datatypes/String.h>
 #include <Core/Matlab/matlabarray.h>
 #include <Core/Matlab/matlabconverter.h>
-#include <Core/Datatypes/MatrixTypeConversions.h>
+#include <Core/Python/PythonDatatypeConverter.h>
+#include <Dataflow/Network/ModuleStateInterface.h>
+#include <boost/variant/apply_visitor.hpp>
 
 using namespace SCIRun;
-using namespace SCIRun::Core::Python;
 using namespace SCIRun::Core::Algorithms;
 using namespace SCIRun::Core::Datatypes;
+using namespace SCIRun::Core::Python;
+using namespace SCIRun::Dataflow::Networks;
 using namespace SCIRun::MatlabIO;
 
+namespace py = boost::python;
+
 namespace
 {
+  struct PythonObjectVisitor : boost::static_visitor<py::object>
+  {
+    py::object operator()(const int& v) const { return py::object(boost::get<int>(v)); }
+    py::object operator()(const std::string& v) const { return py::object(boost::get<std::string>(v)); }
+    py::object operator()(const double& v) const { return py::object(boost::get<double>(v)); }
+    py::object operator()(const bool& v) const { return py::object(boost::get<bool>(v)); }
+    py::object operator()(const AlgoOption& v) const
+    {
+      THROW_INVALID_ARGUMENT("Conversion to AlgoOption is not implemented.");
+    }
+    py::object operator()(const VariableList& v) const
```

Comment:
What is AlgoOption? Is it convertible to a Python object?

Reply:
It's just an enumerated string value

Reply:
Is this like an c++ enum but with string values?

Reply:
It's just a string but there's a list of possible string values also attached to it; just look at the class